src/operations/blas3/gemm_no_local.hpp

```diff
@@ -416,19 +444,28 @@ class Gemm<input_t, output_t, DoubleBuffer, NbcA, NbcB, ClSize, tile_type,
    */
 
   template <index_t item_size, index_t next_element, bool check_block,
-            typename PointerType, typename check_boundary>
+            index_t work_per_load, typename PointerType,
+            typename check_boundary>
   SYCL_BLAS_INLINE void load(PointerType ptr, element_t *reg, const index_t &ld,
                              index_t index, const check_boundary &chk_boundary,
                              const bool out_of_range) noexcept {
     if (out_of_range) {
       return;
     }
 #pragma unroll
-    for (int i = 0; i < item_size; i++) {
-      reg[i] =
-          do_check<check_block>(chk_boundary(index)) ? ptr[0] : element_t(0);
+    for (int i = 0; i < item_size / work_per_load; i++) {
+      bool in_range =
+          do_check<check_block>(chk_boundary(index + (work_per_load - 1)));
```

Comment:
Would it be good to add a comment or something explaining what you are dong here?

Reply:
Good idea, added some comments to this function!